models/WebsiteSetting.php

```diff
@@ -274,6 +278,26 @@ public function delete(): void
             unset(self::$nameIdMappingCache[$nameCacheKey]);
         }
 
+        $event = new WebsiteSettingEvent($this);
+
+        $this->dispatchEvent($event, WebsiteSettingEvents::PRE_DELETE);
+
         $this->getDao()->delete();
+
+        $this->dispatchEvent($event, WebsiteSettingEvents::POST_DELETE);
+    }
+
+    public function save(): void
+    {
+        $event = new WebsiteSettingEvent($this);
+        $isAdd = $this->id === null;
+
+        $this->dispatchEvent($event, $isAdd ? WebsiteSettingEvents::PRE_ADD : WebsiteSettingEvents::PRE_UPDATE
+        );
+
+        $this->getDao()->save();
+
+        $this->dispatchEvent($event, $isAdd ? WebsiteSettingEvents::POST_ADD : WebsiteSettingEvents::POST_UPDATE
+        );
```

Comment:
Sorry, I broke the formatting

```suggestion
        $this->dispatchEvent($event, $isAdd ? WebsiteSettingEvents::PRE_ADD : WebsiteSettingEvents::PRE_UPDATE);

        $this->getDao()->save();

        $this->dispatchEvent($event, $isAdd ? WebsiteSettingEvents::POST_ADD : WebsiteSettingEvents::POST_UPDATE);
```